configure.in

```
@@ -655,6 +673,14 @@ if test "$GCC" = yes -a "$ICC" = no; then
   if test -n "$NOT_THE_CFLAGS"; then
     CFLAGS="$CFLAGS -Wno-stringop-truncation"
   fi
+  # Greenplum's ORCA translator includes a lot of C headers, and Clang will
+  # warn us about the use of the "register" keyword in Postgres' inline
+  # functions which we can do nothing about. Silence it.
+  NOT_THE_CXXFLAGS=""
+  PGAC_PROG_CXX_VAR_OPT(NOT_THE_CXXFLAGS, [-Wdeprecated-register])
```

```suggestion
  PGAC_PROG_VARCXX_VARFLAGS_OPT(CXX, NOT_THE_CXXFLAGS, [-Wdeprecated-register])
```

Like that? Yeah that's sort of my plan about how to reconcile this in the v12 merge once this lands in Greenplum master.

OTOH it's a bit regrettable we didn't have something like `PGAC_PROG_CXX_VAR_OPT` in upstream, because they haven't found a need to use negative warning flags yet (for C++)

---

OK.

---

After the v12 merge, if we're lucky we might be able to rearrange the headers so that the C++ code doesn't need to pull in any headers that use the deprecated `register` keyword.

---

Oh really? I can add a FIXME comment here, will that serve as a better reminder?

Meanwhile I'm really curious what changes in upstream Postgres did you see as allowing that (any mailing list discussions or patches to point my attention to)? The LLVM JIT code is much better isolated than the ORCA translator, I'm under the impression.

---

@hlinnaka  I did a quick search in recent commits in Upstream Postgres and didn't find anything that seem to help when v12 merges. Can you point me at what you were alluding to? I'm holding off adding a FIXME for now